pkg/protocol/engine/eviction/state.go

```diff
@@ -102,22 +92,65 @@ func (s *State) ActiveRootBlocks() map[iotago.BlockID]iotago.CommitmentID {
 		})
 	}
 
+	// We include genesis as a root block if the start of our active window is the genesis slot.
+	if startSlot == s.settings.APIProvider().APIForSlot(s.lastCommittedSlot).ProtocolParameters().GenesisSlot() {
```

Comment:
We should check if this is really necessary, because initially we have the genesis rootblock in the snapshot.
What could happen is that it gets pruned, but pruning should probably be avoided until we don't need genesis anymore.

Reply:
Yep, the genesis root block gets injected into the corresponding storage slot and pruning should not happen until way past the finalized slot.